### ISSUE TITLE
Python: Support locally run models

### ISSUE THREAD

**Issue by silarsis** [microsoft/semantic-kernel#4527]
Trying to use semantic kernel to reference a locally run LLM (via LLM Studio) and am blocked on two fronts. Using an "OpenAIChatCompletion" it is hard-coded to use the OpenAI API directly, not any other endpoint. Using the "AzureChatCompletion" it's hard-coded to require https, not http.

**To Reproduce**
Calling "kernel.add_chat_service()" with either AzureChatCompletion or OpenAIChatCompletion, attempting to reference a non-Azure non-OpenAI-hosted model fails.

**Expected behavior**
I'd like to be able to specify the OPENAI_API_BASE for the OpenAIChatCompletion so it can reference a different model. Adding this as an optional parameter to OpenAIChatCompletion would be ideal.

**Comment by mennolaan** [comment]
Thank you

> Hi @silarsis and @mennolaan. A lot of our current focus has been to support OpenAI/AzureOpenAI, hence the provided chat/text/embedding completion code you find today. We do allow for users to pass in custom clients for both OpenAI and AzureOpenAI. [Here](https://github.com/microsoft/semantic-kernel/blob/7c3a01c1b6a810677d871a36a9211cca0ed7fc4d/python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion.py#L48C9-L48C21) is an example of a constructor that allows for a custom OpenAI client.
> 
> With that said, there's a PR out to support ollama, which allows users to run LLMs locally. You can find it [here](https://github.com/microsoft/semantic-kernel/pull/4477).
> 
> We also have a [Hugging Face connector](https://github.com/microsoft/semantic-kernel/blob/7c3a01c1b6a810677d871a36a9211cca0ed7fc4d/python/semantic_kernel/connectors/ai/hugging_face/services/hf_text_completion.py#L22C17-L22C17) that supports text/embedding completions.
> 
> Do any of these help your use case?
 Thank you, ill follow this pr to see what comes out of it or that i can start using bits of it prematurely. Allthough im c# focussed, but if the python pr is done i reckon c# will follow shortly.

**Comment by silarsis** [comment]
Confirmed the custom OpenAI client example led me to being able to do what I wanted. Not sure if there's a good place in the docs to link that one in? :)

Thanks for the assistance, much appreciated :)

**Comment by moonbox3** [comment]
Thanks for the confirmation, @silarsis. That's great to hear. I will create a new issue so we can track updating the docs.
